docs/jotai/api/utils.mdx

```diff
@@ -222,6 +222,35 @@ import deepEqual from 'fast-deep-equal'
 const fooFamily = atomFamily((param) => atom(param), deepEqual)
 ```
 
+### TypeScript
+
+The atom family types will be inferred from initializeAtom.
+It's generally recommended to infer types from initializeAtom.
+You can explicitly declare the parameter, value, and atom's update function types using TypeScript generics.
+
+```ts
+atomFamily<Param, Value, Update>(initializeAtom: (param: Param) => WritableAtom<Value, Update>, areEqual?: (a: Param, b: Param) => boolean)
+atomFamily<Param, Value>(initializeAtom: (param: Param) => Atom<Value>, areEqual?: (a: Param, b: Param) => boolean)
+```
+
+Here's a typical usage with a primitive atom.
+
+```ts
+import type { PrimitiveAtom } from 'jotai'
+
+const myFamily = atomFamily((id: number) => atom(id))
+// the type of atom(id) is PrimitiveAtom<number>
+// PrimitiveAtom<number> is WritableAtom<number, SetStateAction<number>>
+```
```

Comment:
```suggestion
The atom family types will be inferred from initializeAtom. Here's a typical usage with a primitive atom.

```ts
import type { PrimitiveAtom } from 'jotai'

/** 
 * here the atom(id) returns a PrimitiveAtom<number>
 * and PrimitiveAtom<number> is a WritableAtom<number, SetStateAction<number>>
 */ 
const myFamily = atomFamily((id: number) => atom(id))
```

You can explicitly declare the type of parameter, value, and atom's setState function using TypeScript generics.

```ts
atomFamily<Param, Value, Update>(initializeAtom: (param: Param) => WritableAtom<Value, Update>, areEqual?: (a: Param, b: Param) => boolean)
atomFamily<Param, Value>(initializeAtom: (param: Param) => Atom<Value>, areEqual?: (a: Param, b: Param) => boolean)
```
```

Reply:
looks like Github has a bug when adding code inside suggestions. I had to artificially escape "```".

Reply:
Thanks. OK, I'll commit it and fix it.